Editor/src/Windows/WelcomeWindow.cpp

```diff
@@ -78,179 +93,193 @@ namespace Nuake
 		ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(32.0f, 32.0f));
 		ImGui::Begin("Welcome Screen", 0, ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoResize);
 		{
+			// Draw Nuake logo
 			{
-				UIFont boldfont = UIFont(Fonts::Title);
-
-				std::string text = "Nuake Engine";
-				auto windowWidth = ImGui::GetWindowSize().x;
-				auto textWidth = ImGui::CalcTextSize(text.c_str()).x;
-
-				ImGui::SetCursorPosX((windowWidth - textWidth) * 0.5f);
-				ImGui::Text(text.c_str());
+				const Vector2 logoSize = _NuakeLogo->GetSize();
+				const ImVec2 imguiSize = ImVec2(logoSize.x, logoSize.y);
+				ImGui::Image((ImTextureID)_NuakeLogo->GetID(), imguiSize, ImVec2(0, 1), ImVec2(1, 0));
 			}
-			{
-				UIFont boldfont = UIFont(Fonts::SubTitle);
-				std::string text = "An IdTech inspired game engine";
-				auto windowWidth = ImGui::GetWindowSize().x;
-				auto textWidth = ImGui::CalcTextSize(text.c_str()).x;
 
-				ImGui::SetCursorPosX((windowWidth - textWidth) * 0.5f);
-				ImGui::Text(text.c_str());
-			}
-			ImGui::Separator();
+			// Add padding under logo
+			ImGui::Dummy(ImVec2(10, 25));
 			{
 				UIFont boldfont = UIFont(Fonts::SubTitle);
-				ImGui::Text("Projects recently opened");
+				ImGui::Text("Open recent");
 			}
 
-			ImVec2 projectsWindowSize = ImGui::GetContentRegionAvail();
-			projectsWindowSize.x *= 0.6f;
+			DrawRecentProjectsSection();
+			ImGui::SameLine();
+			DrawRightControls();
+		}
 
-			ImGui::BeginChild("Projects", projectsWindowSize, true);
-			{
-				const uint32_t itemHeight = 120;
+		ImGui::End();
+		ImGui::PopStyleVar();
+		ImGui::PopStyleVar();
+	}
 
-				for (uint32_t i = 0; i < std::size(_Projects); i++)
-				{
-					ProjectPreview& project = _Projects[i];
+	void WelcomeWindow::DrawRecentProjectsSection()
+	{
+		// Recent projects section takes up 80% of the width
+		ImVec2 projectsWindowSize = ImGui::GetContentRegionAvail();
+		projectsWindowSize.x *= 0.8f;
 
-					float cursorYStart = ImGui::GetCursorPosY();
+		ImGui::BeginChild("Projects", projectsWindowSize, true);
+		{
+			for (uint32_t i = 0; i < std::size(_Projects); i++)
+			{
+				DrawProjectItem(i);
+			}
 
-					std::string selectableName = "##" + std::to_string(i);
-					bool isSelected = SelectedProject == i;
-					if (ImGui::Selectable(selectableName.c_str(), isSelected, ImGuiSelectableFlags_AllowItemOverlap, ImVec2(ImGui::GetContentRegionAvailWidth(), itemHeight)))
+			const float itemHeight = 120.0f;
+			if (ImGui::Button("Import an existing project", ImVec2(ImGui::GetContentRegionAvailWidth(), itemHeight)))
+			{
+				const std::string path = FileDialog::OpenFile("Project file |*.project");
+				if (path != "" && String::EndsWith(path, ".project"))
+				{
+					// Prevent importing the same project twice in the list
+					bool alreadyContainsProject = false;
+					for (auto& p : _Projects)
 					{
-						SelectedProject = i;
+						if (p.Path == path)
+						{
+							alreadyContainsProject = true;
```

Comment:
Once set to true, we could exist right away the for loop
```suggestion
							alreadyContainsProject = true;
							break;
```